lib/tls.js

```diff
@@ -1266,6 +1307,30 @@ exports.connect = function(/* [port, host], options, cb */) {
 
   options.secureOptions = crypto._getSecureOptions(options.secureProtocol,
                                                    options.secureOptions);
+  
+  var isPsk = (options.pskIdentity && options.pskKey) ? true : false;
```

Comment:
Although I prefer the way you wrote it I think the idiomatic way to write this in Javascript is:

```Javascript
var isPsk = !!(options.pskIdentity && options.psKey);